Library/Homebrew/utils.rb

```diff
@@ -26,12 +26,25 @@ def self.require?(path)
 
   # Need to keep this naming as-is for backwards compatibility.
   # rubocop:disable Naming/PredicateMethod
-  def self._system(cmd, *args, **options)
+  sig {
+    params(
+      cmd:     T.any(NilClass, Pathname, String, [String, String], T::Hash[String, T.nilable(String)]),
+      argv0:   T.any(NilClass, Pathname, String, [String, String]),
+      args:    T.any(Pathname, String),
+      options: T.untyped,
+      _block:  T.nilable(T.proc.void),
+    ).returns(T::Boolean)
+  }
+  def self._system(cmd, argv0 = nil, *args, **options, &_block)
     pid = fork do
       yield if block_given?
       args.map!(&:to_s)
       begin
-        exec(cmd, *args, **options)
+        if argv0
```

Comment:
The latest, squashed push adds this condition, because we cannot pass a nil `argv0`: https://docs.ruby-lang.org/en/3.4/Kernel.html#method-i-exec

I'm not sure if this discovery merits rethinking how I've introduced `argv0` throughout the chain from the `Kernel` extensions down to here. I think that I still favor matching the upstream Kernel.exec [sig](https://github.com/sorbet/sorbet/blob/b4092efe0a4489c28aff7e1ead6ee8a0179dc8b3/rbi/core/kernel.rbi#L3033-L3041) though.

 Only a single test of the `stackprof` command hit this path, which leaves me with some nervousness in rolling out this PR. I recommend holding merging until after we've cut the latest release.